broadinstitute/single_cell_portal_core
# Add file upload for Anndata (.h5ad) and Seurat data to wizard behind Feature flag (SCP-4525)

Changes:
- Added a new section to the upload wizard for the non-visualizable files
- Added sections for uploading Seurat Data files and H5ad Anndata files
- Rename the Miscellaneous/other section to be just Miscellaneous 
- Added a migration and feature flag to configure the ability to add the H5ad & Seurat files
- Added test for the Seurat Data files and H5ad Anndata files utilizing the style of test for the reference image test

Technical Notes:
- Currently there is a bit of code duplication between the 3 non-visualizable file types components, however, this was done with the idea being that moving forward these will become more different than same when we add validation etc. 
- I toyed with the idea of doing a migration of current files that were uploaded as "Other" that have the h5ad or seurat data file extensions to the new categories however I would like to get more feedback from folks on if they think this is a good idea or not. Also since the feature flag currently hides the H5ad/Seurat sections would need to wait until we made those sections GA.


To test:
- Pull this branch
- Run the DB migration with: `bin/rails db:migrate`
- Start up the Rails server and navigate to localhost
- Start up the delayed jobs in a separate terminal with `./rails_local_setup.rb && source config/secrets/.source_env.bash && bin/rails jobs:work`

- Go to a study that you own
- Go to the upload wizard
- You should see the new section for all the non-viz files
- You should only see the Miscellaneous file type in that section

- Go to the feature flags section in the portal
- Set the feature flag `h5ad_and_seurat_upload` to true for your user
- Go back to the upload wizard of the study you own 
- You should now see the Seurat Data and Anndata (.h5ad) sections available
- Go to the new sections and try to upload appropriate files

If you go to the HCA site: https://data.humancellatlas.org/explore/projects
you can filter by file type and download data to try uploading

For easy testing for the Anndata (.h5ad) you will want a file with extention `.h5ad `and for the Seurat data a file with extension `.rds`

![Screen Shot 2022-07-27 at 5 56 52 PM](https://user-images.githubusercontent.com/54322292/181599066-2fa6fa6e-dc6f-4e3e-a50d-d55de1cad9fc.png)
![Screen Shot 2022-07-27 at 5 57 03 PM](https://user-images.githubusercontent.com/54322292/181599068-8f6cc225-4a6e-4138-87ec-40e87305a916.png)
